internal/terraform/context_apply_deferred_test.go

```diff
@@ -2589,6 +2589,36 @@ import {
 			},
 		},
 	}
+
+	unknownImportDefersConfigGeneration = deferredActionsTest{
+		configs: map[string]string{
+			"main.tf": `
+variable "id" {
+	type = string
+}
+
+import {
+	id = var.id
+	to = test.a
+}
+`,
+		},
+		stages: []deferredActionsTestStage{
+			{
+				buildOpts: func(opts *PlanOpts) {
```

Comment:
Don't we want to add assertions on the generated code?

Reply:
We don't actually generate code this time around. If the ID is unknown then we can't get any information about the target resource anyway so we'd have nothing to generate. The follow up plan where the Id is known is (hopefully) where the new config will be generated. That has dedicated tests around it already.

Reply:
It would probably be noisy to generate a comment noting that we don't import this time around but in the next plan / apply cycle, right?

Reply:
that information is available in the plan, so when we add this to the plan renderer we'll include a comment like that. For now, none of the deferred actions are mentioned in the plan at all so this resource is skipped like the rest of them.